group.go

```diff
@@ -0,0 +1,54 @@
+package goload
+
+import (
+	"context"
+	"github.com/mroth/weightedrand/v2"
+	"github.com/rs/zerolog/log"
+	"time"
+)
+
+type executorGroup struct {
+	name    string
+	chooser *weightedrand.Chooser[Executor, int]
+	weight  int
+	timeout time.Duration
+}
+
+// TODO: add options
+func NewGroup(name string, weight int, executors []Executor) Executor {
+	if len(executors) == 0 {
+		log.Fatal().Msg("group can't be empty")
+	}
+	choises := make([]weightedrand.Choice[Executor, int], 0, len(executors))
+	for _, exec := range executors {
+		choises = append(choises, weightedrand.NewChoice(exec, exec.Options().Weight))
+	}
+	chooser, err := weightedrand.NewChooser(
+		choises...,
+	)
+	if err != nil {
+		log.Fatal().Err(err).Msg("can't create chooser")
+	}
+
+	return &executorGroup{
+		name:    name,
+		chooser: chooser,
+		weight:  weight,
```

Comment:
Shouldn't the weight be a summed up version of each executors weight?

Reply:
Hmm good question. The way we currently implement this in our loadtest it wouldn't make sense. We use the group to duplicate tests across shopIds:
``` golang
func NewShopIdGroup(shopIds []int64, name string, weight int, f func(int64) goload.Executor) goload.Executor {
	targets := sliceutils.Map(shopIds, func(shopId int64) goload.Executor {
		return f(shopId)
	})

	return goload.NewGroup(name, weight, targets)
}
```

But since we use this on every endpoint it would cancel out 🤔

Reply:
Changed it so by default it's the sum but you can override it with an option